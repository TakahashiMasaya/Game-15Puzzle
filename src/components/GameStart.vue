<template>
  <div class="game-start">
    <p>15 Puzzle Game</p>
    <button @click.stop="start">Press Start!!</button>
  </div>
</template>

<script lang="ts" setup>
const props = defineProps<{
  start: () => void;
}>();
</script>

<style lang="scss" scoped>
.game-start {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  height: 100%;
  width: 100%;
  p {
    font-size: 4rem;
    font-weight: bold;
    line-height: 4rem;
    margin-bottom: 30px;
  }
  button {
    display: inline-block;
    border: 3px solid var(--vt-c-text-dark-2);
    background-color: rgba(255, 255, 255, 0.2);
    color: var(--vt-c-text-dark-2);
    padding: 20px;
    border-radius: 5px;
    font-size: 2rem;
    cursor: pointer;
  }
}
</style>
