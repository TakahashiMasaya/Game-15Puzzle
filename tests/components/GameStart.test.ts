import { mount } from '@vue/test-utils';
import GameStart from '@/components/GameStart.vue';

describe('GameStart', () => {
  let clickCount = 0;
  const wrapper = mount(GameStart, {
    props: {
      start: () => {
        clickCount += 1;
      },
    },
  });
  it('snapshots', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('click test', async () => {
    await wrapper.find('button').trigger('click');
    expect(clickCount).toBe(1);
  });
});
