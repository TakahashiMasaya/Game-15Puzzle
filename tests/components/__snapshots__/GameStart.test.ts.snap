// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameStart snapshots 1`] = `
<div
  class="game-start"
>
  <p>
    15 Puzzle Game
  </p>
  <button>
    Press Start!!
  </button>
</div>
`;
